### PR TITLE
Regex fix & options UI

### PR DESCRIPTION
**Pull request by oby1**
Don't know if the language on the page has changed or if it differs for different users but I had to change the regex slightly to match the text on the page.

Also added an options UI to control the refresh interval.  Should come in handy for additional options like "complete checkout"

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+chrome.runtime.onInstalled.addListener(function() {
+        const interval_seconds = 30
+	chrome.storage.sync.set({refresh_interval: interval_seconds}, function() {
+		console.log(`The refresh interval is ${interval_seconds} seconds`);
```
Comment (oby1) [inline background.js]:
```suggestion
        const refreshInterval = 30
	chrome.storage.sync.set({refreshInterval: refreshInterval}, function() {
		console.log(`The refresh interval is ${refreshInterval} seconds`);
```

Too much Python has me defaulting to snake_case these days.

File: content.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 
+
+
 const init = () => {
   const entireBody = document.body.innerText;
   // So it can match one of those messages: 
     // - "No delivery windows available", 
     // - "No attended delivery windows are available"
     // - "No doorstep delivery windows are available";   
-  if (entireBody.search(/No(.*)delivery windows(.*)are available/) !== -1) {
```
Comment (oby1) [inline content.js:9]: Screen Shot 2020-04-11 at 11 34 14 PM

File: content.js [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
-    setTimeout(() => {
-      window.location.reload();
-    }, 30000);
+
+    chrome.storage.sync.get('refresh_interval', function(data) {
+      console.log(`waiting ${data.refresh_interval} seconds`);       
+
+      setTimeout(() => {
+        window.location.reload();
+      }, data.refresh_interval * 1000);
```
Comment (oby1) [inline content.js]:
```suggestion
    chrome.storage.sync.get('refreshInterval', function(data) {
      console.log(`waiting ${data.refreshInterval} seconds`);       

      setTimeout(() => {
        window.location.reload();
      }, data.refreshInterval * 1000);
```

File: options.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Reload Amazon Fresh Options</title>
+</head>
+<body>
+  Refresh Interval (seconds):
+  <input id="refresh_interval" type="number"/>
```
Comment (oby1) [inline options.html]:
```suggestion
  <input id="refresh-interval" type="number"/>
```

File: options.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+function save_options() {
+  const refresh_interval = document.getElementById('refresh_interval').value;
+
+  chrome.storage.sync.set({refresh_interval: refresh_interval}, function() {
```
Comment (oby1) [inline options.js]:
```suggestion
  const refreshInterval = document.getElementById('refresh-interval').value;

  chrome.storage.sync.set({refreshInterval: refreshInterval}, function() {
```

File: options.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    const status = document.getElementById('status');
+    status.textContent = 'Options saved.';
+    setTimeout(function() { status.textContent = ''; }, 750);
+  });
+}
+
+
+function restore_options() {
+  chrome.storage.sync.get({refresh_interval: 30}, function(data) {
+    document.getElementById('refresh_interval').value = data.refresh_interval;
```
Comment (oby1) [inline options.js]:
```suggestion
  chrome.storage.sync.get({refreshInterval: 30}, function(data) {
    document.getElementById('refresh-interval').value = data.refreshInterval;
```

File: content.js (hunk 2 above)
Comment (lastquestion) [reply content.js:9]: I think there's some crazy A/B testing going on. I still get the original text. This is pretty good.

I was thinking what I actually wanted to do was search for buttons that match "actual time slots", but I didn't have a proper body saved of actual availability when I first wrote the script...

File: content.js [hunk 7]
```diff
@@ -1,17 +1,23 @@
...
     // - "No doorstep delivery windows are available";   
-  if (entireBody.search(/No(.*)delivery windows(.*)are available/) !== -1) {
+  if (entireBody.search(/No(.*)delivery windows(.*)available/) !== -1) {
     console.log('no avail');
-    console.log('waiting 30 seconds');
-    setTimeout(() => {
-      window.location.reload();
-    }, 30000);
+
+    chrome.storage.sync.get('refreshInterval', function(data) {
```
Comment (lastquestion) [inline content.js:14]: 👏

**Review by lastquestion — APPROVED** [review]
Thanks for your contribution!

I managed to snag a slot today so I am going to try to actually click on buttons. I'll add an option "beta" or something and under that, it'll attempt to click through. I was dreading adding an options page. so thanks 💯 

Also, I'll probably add prettier.

**Pull request merged**